src/unix/joystick.cpp

```diff
@@ -178,15 +178,19 @@ void* wxJoystickThread::Entry()
 
             if ( (j_evt.type & JS_EVENT_BUTTON) && (j_evt.number < wxJS_MAX_BUTTONS) )
             {
+                // Bounds check for bit shift in this block.
+                wxCHECK_MSG(j_evt.number >= 0, (void *) -1, "Shift count negative.");
```

Comment:
I'm not sure if it's really the best idea to exit the joystick thread completely if we get invalid input. Perhaps it would be better to log it using `wxLogDebug()` and ignore it but continue running?

Reply:
OK. I thought this only killed the thread in debug builds. I've removed the check - as in the OSX case, above, the input will be >= 0, anyway.